File: src/SlimShader/InputOutputSignature/RegisterComponentType.cs
using SlimShader.Util;

namespace SlimShader.InputOutputSignature
{
	public enum RegisterComponentType
	{
		Unknown = 0,

		[Description("uint")]
		UInt32 = 1,

		[Description("int")]
		SInt32 = 2,

		[Description("float")]
		Float32 = 3
	}
}

File: src/cpp/Source/Chunks/Xsgn/RegisterComponentType.cpp
#include "PCH.h"
#include "RegisterComponentType.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(RegisterComponentType value)
{
	switch (value)
	{
	case RegisterComponentType::UInt32 :
		return "uint";
	case RegisterComponentType::SInt32 :
		return "int";
	case RegisterComponentType::Float32 :
		return "float";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}